xarray/core/computation.py

```diff
@@ -1968,7 +1968,7 @@ def polyval(
         raise ValueError(
             f"Dimension `{degree_dim}` should be a coordinate variable with labels."
         )
-    if not np.issubdtype(coeffs[degree_dim].dtype, int):
+    if coeffs[degree_dim].dtype.kind not in "i":
```

Comment:
I think we can just change `int` by `np.integer`
So use `np.issubdtype(coeffs[degree_dim].dtype, np.integer)`

Reply:
I guess that should have the same result but with a smaller diff

Reply:
Should I amend the commit to use `np.issubdtype(coeffs[degree_dim].dtype, np.integer)` instead of `obj.dtype.kind`?

Reply:
I would prefer, it is easier to understand when reading the code.